dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py

```diff
@@ -1977,6 +2028,149 @@ def prepare_for_evalfadept_external_call(self, used_functions):
         # Need to dump in reversed order to put the functions called by other functions in first
         for func_body in reversed(list_functions_body):
             self.list_for_evalfadept_external_call.extend(func_body)
+
+    ##
+    # replace function names and double variables by their adept equivalent when required
+    # @param self : object pointer
+    # @param line : line to analyze
+    # are you ready??
```

Comment:
definetely not!

Reply:
I though hard about how to simplify this method but failed.... I didn't fund any other way than parsing the call argument per argument

Reply:
The only other option that comes to my mind would be to use structural infos described somewhere in the generated code to know beforehands where to do replacements (kind of selective replacements). I suppose you already considered this option so I also suppose there is not such infos in the json or other companion files. Perhaps in an internal structure in OMC they have the information - to do their own automatic differentiation maybe? - and we could use this if we know it. I'm not really sure it would be simpler. Anyway, if you think it can be useful, we can try to ask Bielefeld's guys for example.